ddl/ddl_api.go

```diff
@@ -1029,6 +1026,24 @@ func (d *ddl) CreateTableWithLike(ctx sessionctx.Context, ident, referIdent ast.
 	return errors.Trace(err)
 }
 
+func buildTableInfoWithLike(ident ast.Ident, referTblInfo *model.TableInfo) model.TableInfo {
+	tblInfo := *referTblInfo
+	// Check non-public column and adjust column offset.
+	newColumns := referTblInfo.Cols()
+	newIndices := make([]*model.IndexInfo, 0, len(tblInfo.Indices))
+	for _, idx := range tblInfo.Indices {
+		if idx.State == model.StatePublic {
```

Comment:
What will happen when we have not added the non-public column?
Is it possible to miss some columns which are non-public?

Reply:
If a column/index is non-public, then it is invisible to user( client ), So they won't feel miss.